File: src/main/scala/org/xbery/artbeams/products/service/ProductService.scala
package org.xbery.artbeams.products.service

import org.xbery.artbeams.common.context.OperationCtx
import org.xbery.artbeams.products.domain.{EditedProduct, Product}

/**
  * @author Radek Beran
  */
trait ProductService {
  def findProducts(): Seq[Product]

  def saveProduct(edited: EditedProduct)(implicit ctx: OperationCtx): Either[Exception, Option[Product]]

  def findBySlug(slug: String): Option[Product]
}

File: src/main/kotlin/org/xbery/artbeams/products/service/ProductService.kt
package org.xbery.artbeams.products.service

import org.xbery.artbeams.common.context.OperationCtx
import org.xbery.artbeams.products.domain.EditedProduct
import org.xbery.artbeams.products.domain.Product

/**
 * @author Radek Beran
 */
interface ProductService {
    fun findProducts(): List<Product>
    fun saveProduct(edited: EditedProduct, ctx: OperationCtx): Product?
    fun findBySlug(slug: String): Product?
}
